### DISCUSSION TITLE
113

### DISCUSSION THREAD

**Discussion by giscus** [CODe5753/giscus#2] [Comments]
# 113

실습 환경 MySQL 5.7버전 사용 Windows 10 Entity ID 전략은 IDENTITY Java 11 서론 실무에서 MySQL 5.7 버전을 사용하고 있고, JPA Entity의 ID 전략은 IDENTITY를 사용하고 있는데, 이때 JPARepository를 이용한 saveAll과 save가 bulk insert로 처리되지 않는 것을 발견했습니다. 당장은 saveAll로 아주 많은 데이터를 저장할 일이 없다고 하더라도, 왜 이렇게 작동하는지 확인하고 추후에 동일한 환경에서 bulk 연산을 처리해야 할 때 유연하게 대처할 수 있어야 합니다. 따라서, 이번 기회에 상황과 원인을 파악하고 어떻게 대처할 수 있는지 알아보겠습니다. 결론만 보고 싶으시다면 하단의 “정리” 부분을 보시면 되겠습니다. 작동..

https://imksh.com/113

**Comment by hugehoo** [comment]
승현님 넘 잘읽었습니다. 벌크 인서트 검색하러 들어왔다가 pk 얻어오는 과정까지 배울 수 있어 넘 유익했습니다 감사합니다!

**Reply by CODe5753** [reply]
내용의 흐름이 살짝 어색했을텐데 그 와중에도 이해를 하셨다니 대단하십니다 🙌🏻
감사해요 성후님!! 😊

**Comment by Denia-park** [comment]
내용 잘 보고 갑니다!

Bulk로 처리해야 할 때, JPA가 불편하긴 하죠 ㅎㅎ

**Reply by CODe5753** [reply]
@Denia-park 감사합니다 :)
